components/dashboard/src/onboarding/LinkedInBanner.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useQuery } from "@tanstack/react-query";
+import classNames from "classnames";
+import { FC } from "react";
+import { useLinkedIn } from "react-linkedin-login-oauth2";
+import Alert from "../components/Alert";
+import { Button } from "../components/Button";
+import SignInWithLinkedIn from "../images/sign-in-with-linkedin.svg";
+import { getGitpodService } from "../service/service";
+import { LinkedInProfile } from "@gitpod/gitpod-protocol";
+
+type Props = {
+    onSuccess(profile: LinkedInProfile): void;
+};
+export const LinkedInBanner: FC<Props> = ({ onSuccess }) => {
+    const {
+        data: clientID,
+        isLoading,
+        isError,
+    } = useQuery(
+        ["linkedin-clientid"],
+        async () => {
+            return (await getGitpodService().server.getLinkedInClientId()) || "";
+        },
+        { enabled: true },
+    );
+
+    const { linkedInLogin } = useLinkedIn({
+        clientId: clientID || "",
+        redirectUri: `${window.location.origin}/linkedin`,
+        scope: "r_liteprofile r_emailaddress",
+        onSuccess: (code) => {
+            console.log("success", code);
+            getGitpodService()
+                .server.connectWithLinkedIn(code)
+                .then((profile) => {
+                    onSuccess(profile);
+                })
+                .catch((error) => {
+                    console.error("LinkedIn connection failed", error);
+                });
+        },
+        onError: (error) => {
+            console.error("error", error);
+        },
+    });
+
+    return (
+        <>
+            <div
+                className={classNames(
+                    "mt-6 p-6",
+                    "border-2 border-dashed rounded-md space-y-4",
+                    "bg-gray-50 dark:bg-gray-800",
+                )}
+            >
+                <div className="flex items-center justify-center space-x-6">
+                    <span className="text-4xl">🎁</span>
+                    {/* TODO: Shouldn't need a fixed width here, but was hard to center otherwise  */}
+                    <p className="w-64 text-base text-gray-500 dark:text-gray-100">
+                        Receive <strong>500 credits</strong> per month by connecting your <strong>LinkedIn</strong>{" "}
+                        account.
+                    </p>
+                </div>
+                <Button
+                    className="w-full flex items-center justify-center space-x-2"
+                    onClick={linkedInLogin}
+                    disabled={isLoading || !clientID}
+                >
+                    <img src={SignInWithLinkedIn} width={20} height={20} alt="Sign in with Linked In" />
+                    <span>Connect with LinkedIn</span>
+                </Button>
```

Comment:
**question**: Clicking this just redirected me to the next step as @svenefftinge mentioned in https://github.com/gitpod-io/gitpod/pull/17074#issuecomment-1505132748. Is this expected?

Re-posting here [an export from the design specs](https://www.figma.com/file/CW3DlGSqimERKdQ56qjUM6/Allow-users-to-connect-their-LinkedIn-account-during-the-onboarding-%23UX-8?node-id=3227%3A1588&t=L0zhXMh0DTBwjizF-1).

<img width="2560" alt="AllStates" src="https://user-images.githubusercontent.com/120486/231450582-84ac019c-f519-4410-bcc4-29e4ef032ffc.png">

Reply:
Thanks for noticing @gtsiolis!

The auto-forward should now be fixed.

Also,
- only the `Default` spec is implemented right now
- the `Success` spec is not necessary given that success auto-forwards to the next page
- the `Error` spec is not implemented yet (errors are logged into the console instead)
- I'm not sure when the `Skip` spec should be used if at all

Please let me know what exactly is blocking this PR from being merged, or please resolve if nothing is blocking 😁